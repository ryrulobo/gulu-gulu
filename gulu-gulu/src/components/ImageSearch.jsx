export default function ImageSearch({ results }) {
  return (
    <div className="flex flex-wrap justify-center items-center">
      {results.map(({ hostPageUrl, thumbnailUrl, name }, index) => (
        <a
          className="sm:p-3 p-5"
          href={hostPageUrl}
          key={index}
          target="_blank"
          rel="noreferrer"
        >
          <img
            src={thumbnailUrl}
            alt={"name"}
            loading="lazy"
            className="max-w-36 max-h-48 hover:scale-105 ease-in duration-100"
          />
          <div className="hover:underline">
            <p className="text-sm mt-2">
              {name && name.length > 40 ? name.substring(0, 40) : name}
            </p>
            <p className="text-sm">
              {hostPageUrl && hostPageUrl.length > 40
                ? hostPageUrl.substring(0, 40)
                : hostPageUrl}
            </p>
          </div>
        </a>
      ))}
    </div>
  );
}
