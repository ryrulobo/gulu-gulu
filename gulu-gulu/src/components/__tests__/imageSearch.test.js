import { render, cleanup, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import ImageSearch from "../ImageSearch";

afterEach(() => {
  cleanup();
});

const results = [
  {
    hostPageUrl: "test.com",
    thumbnailUrl: "test.com/image.png",
    name: "this is test title",
  },
];

test("Render image search", () => {
  render(<ImageSearch results={results} />);
  const imageSearchElement = screen.getByTestId("imageSearch");
  expect(imageSearchElement).toBeInTheDocument();
  expect(imageSearchElement).toContainHTML("a");
  expect(imageSearchElement).toContainHTML("img");
  expect(imageSearchElement).toContainHTML("p");
  expect(imageSearchElement.children).toHaveLength(1);
});

test("Matches snapshots", () => {
  const tree = renderer.create(<ImageSearch results={results} />).toJSON();
  expect(tree).toMatchSnapshot();
});
