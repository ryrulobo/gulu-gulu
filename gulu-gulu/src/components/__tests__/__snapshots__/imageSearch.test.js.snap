// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshots 1`] = `
<div
  className="flex flex-wrap justify-center items-center"
  data-testid="imageSearch"
>
  <a
    className="sm:p-3 p-5"
    href="test.com"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="name"
      className="max-w-36 max-h-48 hover:scale-105 ease-in duration-100"
      loading="lazy"
      src="test.com/image.png"
    />
    <div
      className="hover:underline"
    >
      <p
        className="text-sm mt-2"
      >
        this is test title
      </p>
      <p
        className="text-sm"
      >
        test.com
      </p>
    </div>
  </a>
</div>
`;
